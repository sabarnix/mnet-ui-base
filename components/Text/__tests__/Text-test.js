"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

require("jest-styled-components");

var _MnetUIBase = require("../../MnetUIBase");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

test('renders', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, null, "text")));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders size', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "xsmall"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "small"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "medium"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "large"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "xlarge"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    size: "xxlarge"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders textAlign', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    textAlign: "start"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    textAlign: "center"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    textAlign: "end"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders margin', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: "small"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: "medium"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: "large"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: "none"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      vertical: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      horizontal: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      bottom: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      top: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      left: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    margin: {
      right: 'small'
    }
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
var LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';
test('renders truncate', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    truncate: false
  }, LONG), /*#__PURE__*/_react["default"].createElement(_.Text, {
    truncate: true
  }, LONG)));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders color', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    color: "status-critical"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders tag', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    as: "div"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('proxies tag', function () {
  var tagComponent = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    tag: "div"
  })));

  var asComponent = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    as: "div"
  })));

  expect(tagComponent.toJSON()).toEqual(asComponent.toJSON());
});
test('renders weight', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Text, {
    weight: "normal"
  }), /*#__PURE__*/_react["default"].createElement(_.Text, {
    weight: "bold"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});