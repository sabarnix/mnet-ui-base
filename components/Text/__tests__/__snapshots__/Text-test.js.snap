// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    text
  </span>
</div>
`;

exports[`renders color 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  color: #e35e59;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
</div>
`;

exports[`renders margin 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c2 {
  margin: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c3 {
  margin: 16px;
  font-size: 12px;
  line-height: 16px;
}

.c4 {
  margin: 0px;
  font-size: 12px;
  line-height: 16px;
}

.c5 {
  margin-top: 4px;
  margin-bottom: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c6 {
  margin-left: 4px;
  margin-right: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c7 {
  margin-bottom: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c8 {
  margin-top: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c9 {
  margin-left: 4px;
  font-size: 12px;
  line-height: 16px;
}

.c10 {
  margin-right: 4px;
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
  <span
    className="c5"
  />
  <span
    className="c6"
  />
  <span
    className="c7"
  />
  <span
    className="c8"
  />
  <span
    className="c9"
  />
  <span
    className="c10"
  />
</div>
`;

exports[`renders size 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 8px;
  line-height: 12px;
}

.c2 {
  font-size: 9.333333333333334px;
  line-height: 13.333333333333334px;
}

.c3 {
  font-size: 12px;
  line-height: 16px;
}

.c4 {
  font-size: 14.666666666666666px;
  line-height: 18.666666666666668px;
}

.c5 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
}

.c6 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
}

<div
  className="c0"
>
  <span
    className="c1"
    size="xsmall"
  />
  <span
    className="c2"
    size="small"
  />
  <span
    className="c3"
    size="medium"
  />
  <span
    className="c4"
    size="large"
  />
  <span
    className="c5"
    size="xlarge"
  />
  <span
    className="c6"
    size="xxlarge"
  />
</div>
`;

exports[`renders tag 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`renders textAlign 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  text-align: left;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  text-align: center;
}

.c3 {
  font-size: 12px;
  line-height: 16px;
  text-align: right;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
</div>
`;

exports[`renders truncate 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
  <span
    className="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
</div>
`;

exports[`renders weight 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  font-weight: normal;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  font-weight: bold;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
</div>
`;
